test/unit/language-extensions/iterable.spec.ts

```diff
@@ -0,0 +1,231 @@
+import * as path from "path";
+import * as util from "../../util";
+import * as tstl from "../../../src";
+import {
+    invalidIterableUse,
+    invalidMultiIterableWithoutDestructuring,
+} from "../../../src/transformation/utils/diagnostics";
+
+const iterableProjectOptions: tstl.CompilerOptions = {
+    types: [path.resolve(__dirname, "../../../language-extensions")],
+};
+
+const testIterable = `
```

Comment:
I get you don't want to repeat youself, but this is not very readable in my opinion. An improvement would be to move every iterable definition you use here closer to the tests that actually use them, maybe it might even be nice to group them with `describe`s